event.go

```diff
@@ -60,7 +60,7 @@ func createTriggerBatchPayload(batch []Event, encryptionKey string) ([]byte, err
 		} else {
 			batch[idx].Data = string(dataBytes)
 		}
-		if len(batch[idx].Data) > maxEventPayloadSize {
+		if len(batch[idx].Data.(string)) > maxEventPayloadSize {
```

Comment:
Also, you should be getting the length of the payload after it has been encoded to `[]byte`, i.e. `len(dataBytes)`.